autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/saving/AutofillSavingCredentialsDialogFragment.kt

```diff
@@ -122,16 +137,33 @@ class AutofillSavingCredentialsDialogFragment : BottomSheetDialogFragment(), Cre
         container: ViewGroup?,
         savedInstanceState: Bundle?,
     ): View {
-        pixelNameDialogEvent(Shown)?.let { pixel.fire(it) }
         autofillFireproofDialogSuppressor.autofillSaveOrUpdateDialogVisibilityChanged(visible = true)
         viewModel.userPromptedToSaveCredentials()
 
         val binding = ContentAutofillSaveNewCredentialsBinding.inflate(inflater, container, false)
         configureViews(binding)
+        observeViewModel()
         return binding.root
     }
 
+    private fun observeViewModel() {
+        viewModel.viewState
+            .flowWithLifecycle(lifecycle, Lifecycle.State.CREATED)
+            .onEach { viewState ->
+                renderViewState(viewState)
+            }.launchIn(lifecycleScope)
+    }
+
+    private fun renderViewState(viewState: ViewState) {
+        keyFeaturesContainer.isVisible = viewState.expandedDialog
+        dialog.takeIf { it is BottomSheetDialog }?.let {
```

Comment:
nit: I think this could be condensed... 

`(dialog as? BottomSheetDialog)?.behavior?.isDraggable = viewState.expandedDialog` should be equivalent i think